src/main/scala/ProducePngPyramid.scala

```diff
@@ -0,0 +1,127 @@
+          .withDefault(14)
+
+        (inputOrcPathOpt, outputPathOpt, minZoomOpt, maxZoomOpt).mapN {
+          (inputOrcPath, outputPath, minZoom, maxZoom) =>
+            val conf =
+              new SparkConf()
+                .setIfMissing("spark.master", "local[*]")
+                .setAppName("World Pop Image Pyramid")
+                .set("spark.serializer", classOf[KryoSerializer].getName)
+                .set("spark.kryo.registrator", classOf[KryoRegistrator].getName)
+                .set("spark.executor.memory", "2g")
+                .set("spark.driver.memory", "2g")
+
+            implicit val ss =
+              SparkSession.builder.config(conf).enableHiveSupport.getOrCreate
+            // val sqlContext = ss.sqlContext
+            ss.withJTS
+
+            // TODO: Parameterize
+            val partitioner = SpatialPartitioner(64)
+
+            val osmRoadsDF: DataFrame = ss.read
+              .schema(BufferedRoadsSchema())
+              .orc(inputOrcPath)
+
+            val countryList = CountryDirectory.countries.map(_._2) // TODO: Replace with full list?
+            val formatter =
+              CountryFormatter(ss.sparkContext, countryList.toList)
+            val countriesRDD: MultibandTileLayerRDD[SpatialKey] =
+              formatter.readCountries
+            val md: TileLayerMetadata[SpatialKey] = countriesRDD.metadata
+            val transform = md.mapTransform
+
+            val osmRoadsRdd: RDD[Feature[JTSGeometry, String]] =
+              osmRoadsDF.rdd.map { geomRow =>
+                val geom = geomRow.getAs[JTSGeometry]("bufferedGeom")
+                val name = geomRow.getAs[String]("countryName")
+                Feature(geom, name)
+              }
+
+            val clippedGeoms: RDD[(SpatialKey, Feature[Geometry, String])] =
+              ClipToGrid(osmRoadsRdd, md.layout)
+
+            val groupedClippedGeoms
+              : RDD[(SpatialKey, Iterable[Feature[Geometry, String]])] =
+              clippedGeoms.groupByKey(partitioner)
+
+            val joinedRdd
+              : RDD[(SpatialKey,
+                     (MultibandTile, Iterable[Feature[Geometry, String]]))] =
+              countriesRDD.join(groupedClippedGeoms, partitioner)
+
+            val options: Rasterizer.Options =
+              Rasterizer.Options(true, PixelIsArea)
+
+            val maskedRdd: RDD[(SpatialKey, Tile)] = joinedRdd.map {
+              case (spatialKey, (tile, features)) => {
+                val unionedGeom: JTSGeometry =
+                  features.map(_.geom).reduce { _ union _ }
+                val tileExtent: Extent = transform(spatialKey)
+                val firstBand = tile.band(0)
+                // TODO: Invert the mask
+                val maskedTile =
+                  firstBand.mask(tileExtent, unionedGeom, options)
+
+                (spatialKey, maskedTile)
+              }
+            }
+
+            val (_, pyramidMetadata) = maskedRdd
```

Comment:
@jbouffard any ideas for this compile error on this line?

```
[error] /Users/andrew/src/geotrellis-road-distance-sdg/src/main/scala/ProducePngPyramid.scala:110:43: No implicit view available from geotrellis.layer.SpatialKey => geotrellis.spark.tiling.TilerKeyMethods[geotrellis.layer.SpatialKey,geotrellis.layer.SpatialKey].
[error]               .collectMetadata[SpatialKey](ZoomedLayoutScheme(WebMercator, 256))
[error]                                           ^
[error] /Users/andrew/src/geotrellis-road-distance-sdg/src/main/scala/ProducePngPyramid.scala:113:60: type mismatch;
[error]  found   : Any
[error]  required: geotrellis.layer.TileLayerMetadata[?]
[error] Error occurred in an application involving default arguments.
[error]               Pyramid.fromLayerRDD(TileLayerRDD(maskedRdd, pyramidMetadata),
[error]                                                            ^
```
If I add an explicit type hint to `pyramidMetadata: TileLayerMetadata[SpatialKey]` I get this instead:
```
[error] /Users/andrew/src/geotrellis-road-distance-sdg/src/main/scala/ProducePngPyramid.scala:109:43: No implicit view available from geotrellis.layer.SpatialKey => geotrellis.spark.tiling.TilerKeyMethods[geotrellis.layer.SpatialKey,geotrellis.layer.SpatialKey].
[error]               .collectMetadata[SpatialKey](ZoomedLayoutScheme(WebMercator, 256))
[error]                                           ^
```

Then the error does not change if I import `geotrellis.spark.tiling`. I'm not sure where else this implicit might be defined.